vignettes/advs.Rmd

```diff
@@ -0,0 +1,491 @@
+
+# Programming Workflow
+
+-   [Metadata](#metadata)
+-   [Initial ADVS Set-up](#advs_start)
+-   [Derive Additional Variables for Anthropometric
+    indicators](#derive_vars)
+-   [Derive Additional Parameters for Anthropometric
+    indicators](#derive_params)
+-   [Remaining ADVS Set-up](#advs_end)
+
+## Metadata {#metadata}
+
+Once the required packages have been loaded, the first step is then to
+prepare the metadata.
+
+```{r message=FALSE}
+library(admiral)
+library(admiralpeds)
+library(dplyr, warn.conflicts = FALSE)
+library(lubridate)
+library(stringr)
+```
+
+Here we have made some default decisions in the package as to which
+metadata to use but you are free to replace these with any other source
+you prefer (as long as you keep the structure of the dataframe
+consistent - as expected by our downstream functions). For example, you
+might want to use a different age range selection of WHO metadata or
+even use International Obesity Task Force standards.
+
+The selection of standard reference files we included and used as
+metadata for this package are as follows:
+
+-   [2000 CDC Growth
+    Charts](https://www.cdc.gov/growthcharts/percentile_data_files.htm)
+-   For BMI only: [2022 CDC Extended BMI-for-age Growth
+    Charts](https://www.cdc.gov/growthcharts/extended-bmi.htm)
+-   [2006 WHO Growth
+    Standards](https://www.who.int/toolkits/child-growth-standards/standards) -
+    Expanded Tables
+
+### Reference Files for Parameters by Age
+
+As growth rates and patterns differ between gender, different reference
+values are used accordingly and for different age groups.
+
+For the applicable parameters here (height/weight/BMI), we made the
+default decision to rely on the WHO standards for children \<2 years of
+age (\<730.5 days) and CDC standards for children \>=2 years of age
+(\>=730.5 days). This is made in accordance with the [CDC
+recommendation](https://www.cdc.gov/growthcharts/who_charts.htm).
+
+So, the first step here needed is to combine the various metadata by
+parameter and ensuring gender and age are shown in a consistent way. We
+do this using the following code for BMI as an example:
+
+```{r eval=TRUE}
+# Default reference sources: WHO for children <2 yrs old (< 730.5 days),
+# and CDC for children >=2 yrs old (>= 730.5 days)
+data(WHO_bmi_for_age_boys)
+data(WHO_bmi_for_age_girls)
+data(cdc_bmiage)
+
+bmi_for_age <- who_bmi_for_age_boys %>%
+  filter(Day < 730.5) %>%
+  mutate(SEX = "M") %>%
+  bind_rows(who_bmi_for_age_girls %>%
+    filter(Day < 730.5) %>%
+    mutate(SEX = "F")) %>%
+  rename(AGE = Day) %>%
+  bind_rows(cdc_bmiage %>%
+    mutate(
+      SEX = case_when(
+        SEX == 1 ~ "M",
+        SEX == 2 ~ "F",
+        TRUE ~ NA_character_
+      ),
+      AGE = AGE * 30.4375
+    )) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
+  mutate(AGEU = "DAYS") %>%
+  arrange(AGE, SEX)
+```
+
+Here is how the combined metadata now looks for the age ranges around 2
```

Comment:
combined metadata for BMI ? at line 136 we are suddenly talking about HDC

Reply:
will try to explain this better to make it clearer 👍